consensus/src/chained_bft/block_storage/block_tree.rs

```diff
@@ -58,7 +56,7 @@ impl<T> LinkableBlock<T> {
 
 impl<T> LinkableBlock<T>
 where
-    T: Serialize + Default + CanonicalSerialize + PartialEq,
+    T: Serialize + Default + PartialEq,
```

Comment:
I'm not super familiar with Serde. I wonder how can we make it explicit and obvious to verify that a struct is using the LCS data format of serde instead of perhaps some other default types?

Reply:
There are two different things here:
1. Serialize/Deserialize which a type implements, allowing it to be serialized/deserialized from *any* format, not just LCS.
2. a Serializer/Deserializer impl for LCS which lets LCS take any type which impls Serialize/Deserialize and convert it to the LCS format.

So you don't need to distinguish between these two things, when you want the LCS format you just use the LCS specific to/from_bytes methods.

Reply:
We talked offline a bit, but for others following along: If you want to serialize to the LCS format you need to use the `lcs::to_bytes()` function explicitly at the time of serialization.